src/geometry/manifold/smooth_manifold_with_corners.lean

```diff
@@ -0,0 +1,445 @@
+/-
+Copyright (c) 2019 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+
+import analysis.calculus.times_cont_diff geometry.manifold.manifold linear_algebra.finite_dimensional
+
+/-!
+# Smooth manifolds (possibly with boundary or corners)
```

Comment:
The implementation notes makes it easier for me to read the general introduction. This is probably bad...